### PR TITLE
Address race-condition when syncing input seeds

### PR DESCRIPTION
**Pull request by chkeita**
## Summary of the Pull Request

_What is this about?_

## PR Checklist
* [x] Applies to work item: #194
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

**Comment by bmc-msft** [comment]
When run as part of an ensemble, this can prevent fuzzer that doesn't find crashes from learning of new inputs from the ensemble.  I have a suggestion for how to handle that.  Let's talk on Monday.

File: src/agent/onefuzz/src/libfuzzer.rs [hunk 1]
```diff
@@ -70,6 +72,15 @@ impl<'a> LibFuzzer<'a> {
             cmd.arg(o);
         }
 
+        // check if a max_time is already set
+        if let None = self
+            .options
+            .iter()
+            .find(|o| o.starts_with("-max_total_time"))
+        {
+            cmd.arg(format!("-max_total_time={}", DEFAULT_MAX_TOTAL_SECONDS));
```
Comment (bmc-msft) [inline src/agent/onefuzz/src/libfuzzer.rs:81]: 👍

File: src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs [hunk 2]
```diff
@@ -96,34 +96,44 @@ impl LibFuzzerFuzzTask {
...
+        let local_input_dir = tempdir()?;
         loop {
-            self.run_fuzzer(worker_id, stats_sender).await?;
+            self.run_fuzzer(&local_input_dir.path(), worker_id, stats_sender)
+                .await?;
+
+            let mut entries = tokio::fs::read_dir(local_input_dir.path()).await?;
+            while let Some(Ok(entry)) = entries.next().await {
+                let destination_path = self.config.inputs.path.clone().join(entry.file_name());
+                tokio::fs::copy(entry.path(), destination_path).await?;
```
[the remark this answers is not in the document]
Comment (chkeita) [reply src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs]: 👍

**Review by bmc-msft — APPROVED** [review]

**Comment by bmc-msft** [comment]
Performing integration testing

**Pull request merged**